libvast/src/system/spawn_index.cpp

```diff
@@ -30,18 +30,17 @@ spawn_index(node_actor::stateful_pointer<node_state> self,
   auto opt = [&](std::string_view key, auto default_value) {
     return get_or(args.inv.options, key, default_value);
   };
-  auto [archive, filesystem, accountant]
+  auto [filesystem, accountant, archive]
     = self->state.registry
-        .find<archive_actor, filesystem_actor, accountant_actor>();
-  if (!archive)
-    return caf::make_error(ec::lookup_error, "failed to find archive actor");
+        .find<filesystem_actor, accountant_actor, archive_actor>();
   if (!filesystem)
     return caf::make_error(ec::lookup_error, "failed to find filesystem actor");
   const auto indexdir = args.dir / args.label;
   namespace sd = vast::defaults::system;
   auto handle = self->spawn(
-    index, static_cast<store_actor>(archive), filesystem, indexdir,
+    index, filesystem, archive, indexdir,
     // TODO: Pass these options as a vast::data object instead.
+    opt("vast.partition-local-stores", sd::partition_local_stores),
```

Comment:
This is missing from `application.cpp`.
I would also prefer to use a string option by the name "default-store" instead. The default value should be "legacy-archive". This way we won't need another option when we introduce other store plugins.

Reply:
Agreed on the broader point, but not sure about the details:
 - "default-store" is a bit misleading because there's no way for the imported data to go to non-default stores.
 - I'm not sure the archive should be "legacy"  and the default value at the same time.

How about `store-backend: archive` as the default?

Reply:
> Agreed on the broader point, but not sure about the details:
> 
>     * "default-store" is a bit misleading because there's no way for the imported data to go to non-default stores.

I expect there to be a way in the future.

> 
>     * I'm not sure the archive should be "legacy"  and the default value at the same time.

Naming it legacy would convey to the user that it is on its way out, but I see your point.